STM32F405RGTx/Core/Src/rtos_threads/measurements_thread.cpp

```diff
@@ -5,12 +5,26 @@
 #include "mc.hpp"
 #include "cmsis_os.h"
 #include "threads.hpp"
+#include "lut.hpp"
 
-// TODO: add defines here for conversions..
+#define CURRENT_SENSE_RESISTANCE                1E-3
+#define ADC_TO_VOLTAGE(adc)                     ( adc * (3.3/(1 << 12)) )
+#define VOLTAGE_TO_CURRENT(voltage)             (voltage / CURRENT_SENSE_RESISTANCE)
+#define VOLTAGE_TO_RESISTANCE(voltage)          ( -10000 / (voltage - 3.3) )
```

Comment:
Realistically, the voltage here will never be 3.3V (or higher) but if there is we will have a division by zero exception. Is it worth adding an assertion before dividing?

Reply:
This conversion should also be done by the LUT. The LUT should do a direct conversion from an ADC value to a temperature.